srv/src/main/java/sap/capire/xtravels/handler/CreationHandler.java

```diff
@@ -6,39 +6,39 @@
 
 import cds.gen.travelservice.Bookings;
 import cds.gen.travelservice.Bookings_;
-import cds.gen.travelservice.TravelService;
 import cds.gen.travelservice.TravelService_;
 import cds.gen.travelservice.Travels;
 import com.sap.cds.ql.Select;
 import com.sap.cds.services.handler.EventHandler;
 import com.sap.cds.services.handler.annotations.Before;
 import com.sap.cds.services.handler.annotations.ServiceName;
+import com.sap.cds.services.persistence.PersistenceService;
 import org.springframework.stereotype.Component;
 
 @Component
 @ServiceName(TravelService_.CDS_NAME)
 class CreationHandler implements EventHandler {
 
-  private final TravelService service;
+  private final PersistenceService db;
 
-  CreationHandler(TravelService service) {
-    this.service = service;
+  CreationHandler(PersistenceService db) {
+    this.db = db;
   }
 
   // Fill in alternative keys as consecutive numbers for new Travels, Bookings, and Supplements.
   // Note: For Travels that can't be done at NEW events, that is when drafts are created,
   // but on CREATE only, as multiple users could create new Travels concurrently.
   @Before(event = EVENT_CREATE)
   void calculateTravelId(final Travels travel) {
-    var result = service.run(Select.from(TRAVELS).columns(t -> t.ID().max().as("maxID")));
+    var result = db.run(Select.from(TRAVELS).columns(t -> t.ID().max().as("maxID")));
     int maxId = (int) result.single().get("maxID");
     travel.setId(++maxId);
   }
 
   // Fill in IDs as sequence numbers -> could be automated by auto-generation
   @Before(event = EVENT_DRAFT_NEW)
   void calculateBookingPos(Bookings_ ref, final Bookings booking) {
-    var result = service.run(Select.from(ref).columns(t -> t.Pos().max().as("maxPos")));
+    var result = db.run(Select.from(ref).columns(t -> t.Pos().max().as("maxPos")));
```

Comment:
I don't think this is correct. The booking positions are aggregated within a travel. Therefore the service is correct, as the ref already restricts to an inactive travel. `PersistenceService` won't be able to handle the draft-related fields.